### DISCUSSION TITLE
Eclipse BPMN to emfcloud?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-emfcloud/emfcloud#86] [All other] [answered]
Hi,

this is a interesting projects and I wonder if it is possible to migrate the [Eclipse BPMN project](https://www.eclipse.org/bpmn2-modeler/) to this platform?
We already have a  [BPMN 2.0 EMF meta model](https://www.eclipse.org/modeling/mdt/?project=bpmn2) which may be a good starting point. But I fear we need to reimplement all the graphical elements. Do you think this is possible with a reasonable effort?

What would be the best way to start?

===
Ralph

**Comment by planger** [comment, marked as answer]
Hi Ralph,

right, one of the goals of EMF.Cloud and GLSP is to support and facilitate the migration of existing EMF-based modeling tools including diagram editors by enabling reasonable reuse. For instance, we created an initial version of the Eclipse Ecore diagram editor, which can imho be considered a migration that is roughly comparable to yours: https://github.com/eclipse-emfcloud/ecore-glsp

As the front-end, however, is based on an entirely different technology stack than the BPMN editor (is it Graphiti?), the frontend needs to be rewritten; that is, the graphical notation needs to be re-implemented based on [GLSP](https://github.com/eclipse-glsp/glsp) (and Sprotty, which is used as an SVG-based rendering framework in GLSP).
The metamodel and parts of the business logic (editing commands, etc.) can likely be reused to large degrees on the GLSP server.

I think I would start with creating a read-only diagram view from your model (and notation if you want to stay compatible) instance based on [GLSP](https://github.com/eclipse-glsp/glsp) and then continue adding editing support incrementally.
